### PR TITLE
Support `ALTER TABLE DROP PRIMARY KEY`

### LINKED ISSUE #591: MySQL "ALTER ... DROP PRIMARY KEY" is not parsed correctly
The following alter statement to drop primary key directly is allowed in MySQL:

```SQL
ALTER TABLE table_name DROP PRIMARY KEY;
```

However, it's not parsed correctly in sqlparser, for example:

```Rust
use sqlparser::dialect::MySqlDialect;
use sqlparser::parser::Parser;
use sqlparser::tokenizer::Token;
use sqlparser::tokenizer::Tokenizer;

#[test]
fn test() {
    let dialect = &MySqlDialect {};
    let sql = r"ALTER TABLE my_table DROP PRIMARY KEY;";
    let mut tokenizer = Tokenizer::new(dialect, sql);
    let tokens: Vec<Token> = tokenizer.tokenize().unwrap();
    let mut parser = Parser::new(tokens, dialect);
    parser.next_token();
    let statement = parser.parse_alter();
    println!("{:?}", statement);
}
```

Which prints 

Ok(AlterTable { name: ObjectName([Ident { value: "my_table", quote_style: None }]), operation: DropColumn { column_name: Ident { value: "PRIMARY", quote_style: None }, if_exists: false, cascade: false } })

Looks like the "drop primary key" is interpreted as "drop column 'PRIMARY'".

### PR DESCRIPTION
**Pull request by ding-young**
## Issue
resolves #591

### CODE REVIEW

File: src/parser.rs [hunk 1]
```diff
@@ -3136,6 +3136,9 @@ impl<'a> Parser<'a> {
                     name,
                     cascade,
                 }
+            } else if self.parse_keyword(Keyword::PRIMARY) {
```
Comment (AugustoFKL) [inline src/parser.rs:3139]: @ding-young `ALTER TABLE tb DROP PRIMARY` isn't allowed and, therefore, shouldn't be acceptable. 

The `let _` in the next line is permissive over this.

Also, I think this is a MySQL-specific statement. Could you please add the validation to accept only `MySQL` and `GenericDialect`, please?

**Review by AugustoFKL — CHANGES_REQUESTED** [review]

**Review by alamb — COMMENTED** [review]
@ding-young  do you have time to implement @AugustoFKL 's suggestion? Otherwise this PR  looks good to me

**Comment by ding-young** [comment]
> @ding-young  do you have time to implement @AugustoFKL 's suggestion? Otherwise this PR  looks good to me

Yes, I'll work on it soon.

**Comment by ding-young** [comment]
@AugustoFKL Thank you for feedback. I changed the code 😃

**Review by AugustoFKL — APPROVED** [review]
LGTM

@alamb ready for approval :)

**Review by alamb — APPROVED** [review]

**Comment by alamb** [comment]
Thanks @ding-young  and @AugustoFKL !

**Pull request merged**